src/qibolab/platforms/utils.py

```diff
@@ -0,0 +1,93 @@
+from types import SimpleNamespace
+
+
+class Channel:
+    """Representation of physical wire connection (channel).
+
+    Name is used as a unique identifier for channels. If a channel
+    with an existing name is recreated, it will refer to the existing object.
+    Channel objects are instantiated by :class:`qibolab.platforms.platform.Platform`,
+    but their attributes are modified and used by instrument designs.
+
+    Args:
+        name (str): Name of the channel as given in the platform runcard.
+
+    Attributes:
+        ports (list): List of tuples (controller (`str`), port (`int`))
+            specifying the QM (I, Q) ports that the channel is connected.
+        qubits (list): List of tuples (:class:`qibolab.platforms.utils.Qubit`, str)
+            for the qubit connected to this channel and the role of the channel.
+        Optional arguments holding local oscillators and related parameters.
+        These are relevant only for mixer-based insturment designs.
+    """
+
+    instances = {}
+
+    def __new__(cls, name):
+        if name is None:
+            return None
```

Comment:
`None` is not an instance of `Channel`, it is extremely weird and counterintuitive to call `Channel` and get a `NoneType` instance instead.